### PR TITLE
Update cmac.h

### PR DESCRIPTION
**Pull request by RosZad**
Minor documentation improvements:
*Standardized file brief description.
*Separated return statements.
*Please verify RFC that I added in file description.

### CODE REVIEW

File: include/mbedtls/cmac.h [hunk 1]
```diff
@@ -93,8 +94,9 @@ int mbedtls_cipher_cmac_starts( mbedtls_cipher_context_t *ctx,
  * \param input         The buffer holding the input data.
  * \param ilen          The length of the input data.
  *
- * \returns             \c 0 on success, or #MBEDTLS_ERR_MD_BAD_INPUT_DATA
- *                      if parameter verification fails.
+ * \returns            \c 0 on success.
+ * \returns            #MBEDTLS_ERR_MD_BAD_INPUT_DATA
```
Comment (hanno-arm) [inline include/mbedtls/cmac.h]: Above, the `MBEDTLS_XXX` constants are not prefixed by `#`; we should make this uniform.

File: include/mbedtls/cmac.h [hunk 2]
```diff
@@ -93,8 +94,9 @@ int mbedtls_cipher_cmac_starts( mbedtls_cipher_context_t *ctx,
...
+ *                     if parameter verification fails.
```
Comment (hanno-arm) [inline include/mbedtls/cmac.h]: Suggestion: Changeto `if the parameter verification fails.`

File: include/mbedtls/cmac.h [hunk 3]
```diff
@@ -1,8 +1,10 @@
 /**
  * \file cmac.h
  *
- * \brief The Cipher-based Message Authentication Code (CMAC) Mode for
- *        Authentication.
+ * \brief This file contains CMAC definitions and functions.
```
Comment (AndrzejKurek) [inline include/mbedtls/cmac.h]: I'd rather have the full name and the abbreviation in the \brief block.

File: include/mbedtls/cmac.h [hunk 4]
```diff
@@ -67,16 +69,15 @@ struct mbedtls_cmac_context_t
  *                      Must be called with an initialized cipher context.
  *
  * \param ctx           The cipher context used for the CMAC operation, initialized
- *                      as one of the following types:<ul>
- *                      <li>MBEDTLS_CIPHER_AES_128_ECB</li>
- *                      <li>MBEDTLS_CIPHER_AES_192_ECB</li>
- *                      <li>MBEDTLS_CIPHER_AES_256_ECB</li>
- *                      <li>MBEDTLS_CIPHER_DES_EDE3_ECB</li></ul>
+ *                      as one of the following types: MBEDTLS_CIPHER_AES_128_ECB,
```
Comment (AndrzejKurek) [inline include/mbedtls/cmac.h]: I'm afraid that this will be less readable than before...

File: include/mbedtls/cmac.h (hunk 1 above)
Comment (RosZad) [reply include/mbedtls/cmac.h]: # is a linking mechanism. The _XXX error codes are a generalization, and therefore cannot be linked. N/A

File: include/mbedtls/cmac.h (hunk 2 above)
Comment (RosZad) [reply include/mbedtls/cmac.h]: No, it is correct the way it is.

File: include/mbedtls/cmac.h (hunk 3 above)
Comment (RosZad) [reply include/mbedtls/cmac.h]: The file brief description should refer to what the file contains. We provide the full name in the detailed description as a service and further clarification.

File: include/mbedtls/cmac.h (hunk 4 above)
Comment (RosZad) [reply include/mbedtls/cmac.h]: I agree, but we discovered an issue with lists and paragrpah breaks in parameter/member documentation in RTF output (completely break tables), which is why bullets were reverted in them. Hopefully future versions of Doxygen will fix the issue, and then we will restore this.

File: include/mbedtls/cmac.h (hunk 2 above)
Comment (hanno-arm) [reply include/mbedtls/cmac.h]: Ok, I'm not a native, I thought there'd need to be an article here, strictly speaking.

**Pull request merged**